src/core/src/energy/global/wolf.rs

```diff
@@ -207,11 +220,23 @@ impl GlobalPotential for Wolf {
 
                 let rij = system.nearest_image(i, j);
                 let force = self.force_pair(info, qi, qj, rij);
-                res[i] += force;
-                res[j] -= force;
+                thread_forces[i] += force;
+                thread_forces[j] -= force;
+            }
+        });
+
+        // At this point all the forces are computed, but the
+        // results are scattered across all thread local Vecs,
+        // here we gather them.
+        let mut forces = vec![Vector3D::zero(); natoms];
+        for mutex in thread_forces_mutex {
+            let thread_forces = mutex.into_inner().unwrap();
```

Comment:
Same here about the unwrap. We should never hit the code, but better have an easy to find error message.